spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -222,12 +222,16 @@ class CometSparkSessionExtensions
      */
     // spotless:on
     private def transform(plan: SparkPlan): SparkPlan = {
-      def transform1(op: UnaryExecNode): Option[Operator] = {
-        op.child match {
-          case childNativeOp: CometNativeExec =>
-            QueryPlanSerde.operator2Proto(op, childNativeOp.nativeOp)
-          case _ =>
-            None
+      def transform1(op: SparkPlan): Option[Operator] = {
+        val allNativeExec = op.children.map {
+          case childNativeOp: CometNativeExec => Some(childNativeOp.nativeOp)
+          case _ => None
+        }
+
+        if (allNativeExec.forall(_.isDefined)) {
+          QueryPlanSerde.operator2Proto(op, allNativeExec.map(_.get): _*)
+        } else {
+          None
```

Comment:
nit: I feel this can become a little easier if we do
```
if (op.children.forall(_.isInstanceOf[CometNativeExec])) {
  ueryPlanSerde.operator2Proto(op, op.children.map(_.asInstanceOf[CometNativeExec].nativeOp): _*)
} else {
  None
}
```

Reply:
Yea